backend/src/v5/services/sso/frontegg/components/users.js

```diff
@@ -66,63 +85,52 @@ Users.triggerPasswordReset = async (email) => {
 	await post(url, { email }, { headers });
 };
 
-Users.uploadAvatar = async (userId, tenantId, formDataPayload) => {
+Users.updateUserDetails = async (userId, { firstName, lastName, profilePictureUrl, ...metadata }) => {
 	try {
 		const config = await getConfig();
-		const headers = {
-			...await getBearerHeader(),
-			...formDataPayload.getHeaders(),
-			[HEADER_ENVIRONMENT_ID]: config.clientId,
-			[HEADER_TENANT_ID]: tenantId,
-			[HEADER_USER_ID]: userId,
-		};
-		const url = `${config.vendorDomain}/team/resources/profile/me/image/v1`;
+		const headers = await getBearerHeader();
+		const url = `${config.vendorDomain}/identity/resources/users/v1/${userId}`;
 
-		const { data } = await put(url, formDataPayload, { headers });
+		const regexSplitForName = /\s(.*)/;
 
-		return data;
+		const user = await Users.getUserById(userId);
+		const [existingFirstName, existingLastName] = user.name?.split(regexSplitForName) ?? ['Anonymous', 'User'];
+
+		const payload = {
+			name: `${firstName || existingFirstName} ${lastName || existingLastName}`.trim(),
+			metadata: JSON.stringify(metadata),
+			profilePictureUrl: profilePictureUrl || user.profilePictureUrl,
+		};
+
+		await put(url, payload, { headers });
 	} catch (err) {
-		throw new Error(`Failed to upload avatar for user(${userId}) from Users: ${err.message}`);
+		throw new Error(`Failed to update user(${userId}) from Users: ${err.message}`);
 	}
 };
 
-Users.updateUserDetails = async (userId, updatedProfile) => {
+Users.uploadAvatar = async (userId, path, options = {}) => {
```

Comment:
This is a bit too high coupling with the `options` being exposed and allowing the callee to do whatever they want there.

I think you exposed it for the mimeType right? just add the mimeType as a optional field here.